### PR TITLE
Issue/5141 order shipment providers

### LINKED ISSUE #5141: Migrate order shipment providers to coroutines
Update `OrderShipmentProvidersRepository`  and replace the use of `Continuation` with calling the suspended functions in FluxC directly (wordpress-mobile/WordPress-FluxC-Android#2173).

### PR DESCRIPTION
**Pull request by 0nko**
Resolves #5141.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepository.kt [hunk 1]
```diff
@@ -55,40 +32,21 @@ class OrderShipmentProvidersRepository @Inject constructor(
...
+        val payload = FetchOrderShipmentProvidersPayload(selectedSite.get(), order)
+        val result = orderStore.fetchOrderShipmentProviders(payload)
+        return when {
+            result.isError -> {
+                WooLog.e(ORDERS, "Error fetching shipment providers : ${result.error.message}")
+                null
             }
-            event.rowsAffected == 0 -> {
+            result.rowsAffected == 0 -> {
                 WooLog.e(ORDERS, "Error fetching shipment providers : empty list")
```
Comment (malinajirka) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepository.kt]: ❓ We might want to consider switching the log level to `info` and not mention "error" in the log. It seems like a valid case when there aren't any providers available, right? Or is that always an error? Wdyt?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    private val siteModel = SiteModel()
+    private val order = OrderTestUtils.generateOrder()
+    private val providers = OrderTestUtils.generateOrderShipmentProviders()
+
+    private lateinit var repository: OrderShipmentProvidersRepository
+
+    @Before
+    fun setup() {
+        repository = OrderShipmentProvidersRepository(selectedSite, orderStore)
+        doReturn(siteModel).whenever(selectedSite).get()
```
Comment (malinajirka) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt]: 🔍 Nitpick: We recently had a discussion on Slack about this. We agreed that `doReturn().whenever()` should be used only in cases when it's necessary and `whenever(..).thenReturn(..)` should be preferred. The main benefit of the latter is that it's type-safe => simplifies refactoring.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        doReturn(order).whenever(orderStore).getOrderByIdentifier(ORDER_IDENTIFIER)
+    }
+
+    @Test
+    fun `When there are shipment providers in local db`() = runBlockingTest {
```
Comment (malinajirka) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt]: 🔍 Nitpick: We might want to consider renaming the tests so the test names mention the expected behavior. It's nice when the reader doesn't need to read the code to understand the behavior of the app -> ideally, they would read just names of unit tests and understand the logic of the screen. Wdyt?

Example: This test could be called for example `given data in local db, when fetch shipment providers invoked, then cached data returned`.

**Review by malinajirka — APPROVED** [review]
Thanks, it looks good to me. I've left a couple nitpicks, but this PR is good to be merged. 🚢

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt (hunk 3 above)
Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/tracking/OrderShipmentProvidersRepositoryTest.kt]: Yeah, that makes sense.

**Pull request merged**